src/misc/ListModel.ts

```diff
@@ -231,16 +231,19 @@ export class ListModel<ElementType extends ListElement> {
 		// We cannot use binary search here because the sort order of items can change based on the entity update and we need to find the position of the
 		// old entity by id in order to remove it.
 
+		// Since every element id is unique and there's no scenario where the same item appears twice but in different lists, we can safely sort just
```

Comment:
Element IDs don't have to be unique, they are not all `GeneratedId`s but that *was* the assumption here in the ListModel that the given IDs don't repeat until the calendar search. Will it work with that?

Reply:
> Element IDs don't have to be unique, they are not all `GeneratedId`s but that _was_ the assumption here in the ListModel that the given IDs don't repeat until the calendar search. Will it work with that?

No, it will not work. Thanks for remembering about this case.
When we generate the further events, they have the same ID as their progenitor

Reply:
It should work, generated events already have the same ID as their progenitor, so do the comparison by the elementIdpart or by the full element will lead to the same behavior.